### PR TITLE
From HackMD plan, part 3: most of the rest of the validation tests

### PR DESCRIPTION
**Pull request by kainino0x**
This one's more messy because I really need to get this done, so I didn't clean up (look for duplicates, organize, etc.) as much. As usual not super necessary to review the plans because we'll look at them again later when doing the TODOs, but if you notice anything lmk.

### CODE REVIEW

**Comment by kainino0x** [comment]
Organizing this is really hard. I keep forgetting where stuff is, not knowing whether we already have something implemented, forgetting that we already have a plan when someone writes new plan/tests, etc.

File: src/webgpu/api/operation/compute_pipeline/entry_point_name.spec.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const description = `
+TODO:
+- Test some weird but valid values for entry point name (both module and pipeline creation
```
Comment (Kangz) [inline src/webgpu/api/operation/compute_pipeline/entry_point_name.spec.ts:3]: nit: also test using one of many entrypoints in the shadermodule, and creating on compute pipeline from each entrypoint in this large shadermodule (and using them?)

File: src/webgpu/api/operation/render_pipeline/entry_point_name.spec.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const description = `
+TODO:
+- Test some weird but valid values for entry point name (both module and pipeline creation
+  should succeed).
```
Comment (Kangz) [inline src/webgpu/api/operation/render_pipeline/entry_point_name.spec.ts:4]: ditto

File: src/webgpu/api/operation/sampling/anisotropy.spec.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+export const description = `
+Tests the behavior of anisotropic filtering.
+
+Note that anisotropic filtering is never guaranteed to occur, but we might be able to test that
+no *more* than the provided maxAnisotropy samples are used, by testing how many unique
+sample values come out of the sample operation.
```
Comment (Kangz) [inline src/webgpu/api/operation/sampling/anisotropy.spec.ts:6]: Or have a test that check the result with and without anisotropy are different, or that aniso is done in the correct direction (by having a 2D gradient and checking we get more of the color in the correct direction). There's a bunch of implementation dependent behavior we'll still want to test, so maybe it is worth having some sort of soft expectations or non-failing expectations for them?

File: src/webgpu/api/validation/buffer/map_state.spec.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export const description = `
+Validation tests for GPUBuffer.mapAsync, GPUBuffer.unmap and GPUBuffer.getMappedRange.
```
Comment (Kangz) [inline src/webgpu/api/validation/buffer/map_state.spec.ts:2]: These are already done in mapping.spec.ts

File: src/webgpu/api/validation/create_pipeline.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+export const description = `
```
Comment (Kangz) [inline src/webgpu/api/validation/create_pipeline.ts:1]: nit: This file is `create_pipeline` while the other one is `createPipeline`. What's the convention?

**Review by Kangz — APPROVED** [review]

File: src/webgpu/api/validation/create_pipeline.ts (hunk 5 above)
Comment (kainino0x) [reply src/webgpu/api/validation/create_pipeline.ts:1]: camelCase is named after a function in the API, but createPipeline is not one (it's createRenderPipeline + createComputePipeline)

File: src/webgpu/api/validation/buffer/map_state.spec.ts (hunk 4 above)
Comment (kainino0x) [reply src/webgpu/api/validation/buffer/map_state.spec.ts:2]: oops, accidentally duplicated

**Pull request merged**